### ISSUE TITLE
UncaughtCorsError Uncaught CORS Error Unknown CORS error  An unknown CORS error occured. The error probably originates from a JavaScript file served from a different origin. (Opening your browser console might give you a hint on the error.)

### ISSUE THREAD

**Issue by haffafdigital** [odoo/odoo#146940]
Impacted versions:

Steps to reproduce:

Current behavior:

Expected behavior:

Video/Screenshot link (optional):

Support ticket number submitted via odoo.com/help (optional):

**Comment by ngist** [comment]
I'm going to add onto this rather than making a new issue, because I think I'm seeing this issue as well. Though the error in my browser is different.

Impacted versions:
16.4 Saas (perhaps others)

Steps to reproduce:
Setup a CDN, as per guide https://www.odoo.com/documentation/17.0/administration/install/cdn.html#:~:text=In%20the%20Odoo%20Website%20app,when%20Developer%20Mode%20is%20activated.

Current behavior:
After loading the page chrome produces the following errrors:
![Screenshot 2023-12-20 134719](https://github.com/odoo/odoo/assets/25966140/30d0acd2-f474-48fe-80d0-f407230497c0)

web.assets_fronend_lazy.js includes two import statements that don't list a URL only URI and since the js is loaded from the CDN ie a CORS request the browser doesn't know how to deal with it, should it use the origin domain or the cdn domain? It chooses the safest option and makes no assumption and sets the domain to about:blank

Expected behavior:
No errors when using the standard CDN setup, 

Video/Screenshot link (optional):

Support ticket number submitted via odoo.com/help (optional):

**Comment by SamekaTV** [comment]
@ngist 
Is excluding /web/assets from CDN settings sufficient to avoid the CORS error and make it work?

**Comment by imunisasi** [comment]
I'm reply this since already issue open for this,
this popup appear in chrome, when opening Technical Views Odoo 17 Enterprise
```
Unknown CORS error

An unknown CORS error occured.
The error probably originates from a JavaScript file served from a different origin.
(Opening your browser console might give you a hint on the error.)
```

Warning :
[Violation] Added non-passive event listener to a scroll-blocking 'mousewheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
ace.js:1436

[Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
ace.js:1436

[Violation] Added non-passive event listener to a scroll-blocking 'touchmove' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
ace.js:1436

Error :
Uncaught ResizeObserver loop completed with undelivered notifications.

**Comment by qxbc** [comment]
> I'm reply this since already issue open for this, this popup appear in chrome, when opening Technical Views Odoo 17 Enterprise
> 
> ```
> Unknown CORS error
> 
> An unknown CORS error occured.
> The error probably originates from a JavaScript file served from a different origin.
> (Opening your browser console might give you a hint on the error.)
> ```
> 
> Warning : [Violation] Added non-passive event listener to a scroll-blocking 'mousewheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952 ace.js:1436
> 
> [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952 ace.js:1436
> 
> [Violation] Added non-passive event listener to a scroll-blocking 'touchmove' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952 ace.js:1436
> 
> Error : Uncaught ResizeObserver loop completed with undelivered notifications.

I have the same problem when I click save after editing the website

**Comment by majorbird-team** [comment]
I have the same issue, started to show recently on V17 enterprise. We did not run an update on the codebase and it started to show up.

**Comment by TheMule71** [comment]
> Uncaught ResizeObserver loop completed with undelivered notifications.

Same here. EE 17.0 on premise. It started showing up, no updates on our side.

**Comment by TheMule71** [comment]
Adding to comment:

![Image](https://github.com/user-attachments/assets/b39a1980-28b9-4235-a734-edc1e1c57a5b)

I wonder if it's something related to the js editor.  Can someone confirm it happens only when loading a page with the ACE editor? (Like, views and other xml editing).

I find it weird its resources are marked "no domain", while all the rest resides under the domain of the web site.

**Comment by imunisasi** [comment]
@TheMule71 , @majorbird-team , @qxbc yes with ace editor outside domain

**Comment by majorbird-team** [comment]
Indeed, the error only happens where ace editor is displayed on pages. I sent an email to our Odoo account manager about this (I believe this is **critical** as it must affect a lot of production dbs out there) - also created a ticket through our enterprise subscription. Will keep you posted if I get feedback.

**Comment by Dg0210** [comment]
I have the same issue today on V17 CC on 8, 9 env servers(even local). We did not have any updates on our code and code base,  does anyone know the reason?

**Comment by ioFilippo** [comment]
Try to see if this fixes the issue, it's working on v15.

``` js
/** @odoo-module **/

import { registry } from "@web/core/registry";
import { UncaughtCorsError } from "@web/core/errors/error_service";

// INFO: this is needed to avoid to intercept CORS errors from external js.
function corsErrorHandler(env, error, originalError) {
    if (error instanceof UncaughtCorsError) {
        const errorsToIgnore = [
            "ResizeObserver loop completed with undelivered notifications.",
            "ResizeObserver loop limit exceeded"
        ]
        if (originalError && errorsToIgnore.includes(originalError)) {
            return true;
        }
    }
}
// INFO: sequence needs to be smallest value so to have priority on the system cors error handler.
errorHandlerRegistry.add("corsErrorToIgnoreHandler", corsErrorHandler, { sequence: 1 });
```

Do not forget to add it into the `__manifest__.py`
``` python
...
    "assets": {
        "web.assets_backend": [
            "module_folder/static/src/js/script.js",
...
```

**Comment by alesisjoan** [comment]
If you can't wait for the PR [](https://github.com/odoo/odoo/pull/194483) you can update chrome to version 133

**Comment by imunisasi** [comment]
> If you can't wait for the [PR](https://github.com/odoo/odoo/pull/194483) you can update chrome to version 133

You save my day @alesisjoan , editing xml in odoo seems annoying must close the warning everytime
